File: pertemuan2/nQueen.cpp
# include <iostream>
using namespace std;

int main() {
    int N; 
    cin >> N;

    char papan[N][N];

    for (int i = 0; i < N;i++) {
        for (int j = 0; j < N; j++) {
            papan[i][j] = '.';
        }
    }

    int nQueen = N;
    if (nQueen == 2 || nQueen == 3) {
        cout<<"Kerajaan tidak dapat dilindungi!"<<endl;
        return 0;
    }
    if (nQueen % 2 == 1) {
        int start = 0;
        int j = 0;
        for (int i = 0; i < N; i++) {
            if (j >= N) { 
                if (start == 2) {
                break;
            }
            start++;
            j = start;
        }
        papan[i][j] = 'Q';
        nQueen--;
        j += 2;
    }
        if (nQueen == 0) {
            for (int i = 0; i < N; i++) {
                for (int j = 0; j < N; j++) {
                    cout <<papan[i][j];
                }
                cout<<endl;
            }
        }else {
            cout<<"Kerajaan tidak dapat dilindungi!"<<endl;
        }
    
    }else {
        int start = 1;
        int j = 1;
        for (int i = 0; i < N; i++) {
            if (j >= N) { 
                if (start == 0) {
                    break;
                }
                start--;
                j = start;
            }
            papan[i][j] = 'Q';
            nQueen--;
            j += 2;
        }
        if (nQueen == 0) {
            for (int i = 0; i < N; i++) {
                for (int j = 0; j < N; j++) {
                    cout <<papan[i][j];
                }
                cout<<endl;
            }
        }else {
            cout<<"Kerajaan tidak dapat dilindungi!"<<endl;
        }
    
    }
    



    return 0;
}

File: pertemuan2/nQueen.py

N = int(input())

posQueen = []
for i in range(N):
    posQueen.append(-1)

row = 0

while (row >= 0) :
    isFound = False

    for column in range(posQueen[row] + 1,N):
        isSafe = True

        for i in range(row):
            isDiagonal = abs(posQueen[i] - column) == abs(i - row)
            if (isDiagonal or posQueen[i] == column) :
                isSafe = False
                break
        if (isSafe):
            posQueen[row] = column
            isFound = True
            break
    if isFound :
        if row == N - 1 :
            break
        else :
            row += 1
            posQueen[row] = -1
    else :
        row -= 1
if row == N - 1:
    for i in range(N):
        for j in range(N):
            if j == posQueen[i]:
                print("Q",end="")
            else :
                print(".",end="")
        print("\n",end="")
else :
    print('Kerajaan tidak dapat dilindungi!')
